nvmeadm/src/nvmf_discovery.rs

```diff
@@ -73,6 +73,19 @@ impl fmt::Display for TrType {
     }
 }
 
+impl FromStr for TrType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "rdma" => Ok(Self::rdma),
+            "tcp" => Ok(Self::tcp),
+            "fc" => Ok(Self::fc),
+            _ => Err("Invalid transport string for TrType".to_string()),
```

Comment:
might be useful to show which value was used
```suggestion
            invalid => Err(format!("Invalid transport type: {invalid}"),
```

Reply:
yeah, displaying `invalid` now. 
On another note- since I've got my hands burned once with dependency changes merging, in this case I don't think the merging of this PR can break anything as this is just a new trait function implementation. Let me know if you think otherwise.